app/models/puppetclass.rb

```diff
@@ -110,8 +107,8 @@ def all_host_ids
     ids.uniq
   end
 
-  def update_total_hosts
-    update_attribute(:total_hosts, all_host_ids.size)
+  def hosts_count
+    all_host_ids.count
```

Comment:
This is full load into memory (pluck, uniq and others), replace with ordinary SQL call here.